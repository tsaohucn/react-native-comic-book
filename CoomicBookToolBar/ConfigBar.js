import React, { Component } from 'react'
import { 
  View, 
  StyleSheet, 
  Text, 
  Animated, 
  Dimensions, 
  TouchableWithoutFeedback ,
  Platform
} from 'react-native'
import { Icon } from 'react-native-elements'
import ComicBookSlider from './ComicBookSlider'
import ComicBookSwitch from './ComicBookSwitch'

export default class ConfigBar extends Component {

  constructor(props) {
    super(props)
  }

  setValue = (value,done) => {
    this.ComicBookSlider.setValue(value,done)
  }

  render() {
    return (
      <Animated.View style={[styles.configBarAnimated,{
        transform: [
          { translateY: this.props.animatedConfigBarY }
        ]}]}
      >
        <View style={styles.configBarRawView}>
          <View style={styles.lightTextView}>
            <Text style={styles.titleText}>亮度調節</Text>
          </View>
          <View style={styles.decagram}>
            <Icon
              name='decagram'
              type='material-community'
              color='white'
            />
          </View>
          <View style={styles.lightSliderView}>
            <ComicBookSlider 
              ref={ref => this.ComicBookSlider = ref}
              style={styles.lightSlider}
              //onSlidingStart={this.onLightSlidingStart}
              onValueChange={this.props.onLightValueChange}
            />
          </View>
          <View style={styles.decagramoutline}>
            <Icon
              name='decagram-outline'
              type='material-community'
              color='white'
            />
          </View>
        </View>

        <View style={styles.configBarRawView}>
          <View style={styles.configBarItem}>
            <Text style={styles.titleText}>閱讀模式</Text>
          </View>
          <View style={styles.readStyle}>
            <Icon
              name='book-open'
              type='material-community'
              color='white'
            />
            <Text style={styles.titleText}>普通模式</Text>
          </View>
          <View style={styles.readStyle}>
            <Icon
              name='book-open-page-variant'
              type='material-community'
              color='white'
            />
            <Text style={styles.titleText}>日漫模式</Text>
          </View>
          <View style={styles.readStyle}>
            <Icon
              name='book-open-variant'
              type='material-community'
              color='white'
            />
            <Text style={styles.titleText}>捲軸模式</Text>
          </View>

          <View style={styles.nonReadStyle}/>
        </View>

        <View style={styles.configBarRawView}>
          <View style={styles.configBarItem}>
            <Text style={styles.titleText}>橫豎屏</Text>
          </View>
          <View style={styles.readStyle}>
            <Icon
              name='desktop-mac'
              type='material-community'
              color='white'
            />
            <Text style={styles.titleText}>橫屏</Text>
          </View>
          <View style={styles.readStyle}>
            <Icon
              name='desktop-classic'
              type='material-community'
              color='white'
            />
            <Text style={styles.titleText}>豎屏</Text>
          </View>

          <View style={styles.nonReadStyle}/>
          <View style={styles.nonReadStyle}/>
        </View>

        <View style={styles.configBarRawView}>
          <View style={styles.configBarItem}>
            <Text style={styles.titleText}>顯示單章節評論</Text>
          </View>
          <ComicBookSwitch/>
        </View>
      </Animated.View>    
    )  
  }
}

const { width, height } = Dimensions.get('window')

const configBarHeight = height*1/2

const styles = StyleSheet.create({
  configBarAnimated: {
    position: 'absolute', 
    height: configBarHeight,
    width,
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
    bottom: 0
  },
  configBarRawView: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 10,
    paddingRight: 10
  },
  configBarItem: {
    flex: 1
  },
  readStyle: {
    flex: 1,
    borderWidth: 1,
    borderColor: 'white',
    marginRight: 5,
    borderRadius: 5,
    height: configBarHeight/5,
    justifyContent: 'center',
    alignItems: 'center'
  },
  nonReadStyle: {
    flex: 1,
    borderWidth: 0,
    borderColor: 'white',
    marginRight: 5,
    borderRadius: 5,
    height: configBarHeight/5,
    justifyContent: 'center',
    alignItems: 'center'
  },
  titleText: {
    color: 'white'
  },
  lightTextView: {
    flex: 2
  },
  decagram: {
    flex: 1,
    paddingTop: 2
  },
  decagramoutline: {
    flex: 1,
    paddingTop: 2
  },
  lightSliderView: {
    flex: 7,
    paddingTop: Platform.OS === 'android' ? 1 : 0
  },
  lightSlider: {
    width: width*4/7
  }
})