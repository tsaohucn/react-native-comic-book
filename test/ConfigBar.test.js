// __tests__/ConfigBar.js
import 'react-native'
import React from 'react'
import ConfigBar from '../CoomicBookTool/ConfigBar'

import renderer from 'react-test-renderer'

jest.mock('react-native-elements', () => {
	return {
		Icon: 'Icon'
	}
})

jest.mock('../CoomicBookTool/ComicBookSlider', () => 'ComicBookSlider')

jest.mock('../CoomicBookTool/ComicBookSwitch', () => 'ComicBookSwitch')

describe('Testing ConfigBar component', () => {

	it('renders correctly with no props', () => {
		const props = {

		}
	  const tree = renderer.create(
	    <ConfigBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

	it('renders correctly with animatedConfigBarY value', () => {
		const props = {
			animatedConfigBarY: 100
		}
	  const tree = renderer.create(
	    <ConfigBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

})