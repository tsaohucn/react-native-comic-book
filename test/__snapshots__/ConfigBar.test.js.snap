// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfigBar component renders correctly with animatedConfigBarY value 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "height": 667,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 100,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        亮度調節
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 2,
        }
      }
    >
      <Icon
        color="white"
        name="decagram"
        type="material-community"
      />
    </View>
    <View
      style={
        Object {
          "flex": 7,
          "paddingTop": 0,
        }
      }
    >
      <ComicBookSlider
        onValueChange={undefined}
        style={
          Object {
            "width": 428.57142857142856,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 2,
        }
      }
    >
      <Icon
        color="white"
        name="decagram-outline"
        type="material-community"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        閱讀模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        普通模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open-page-variant"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        日漫模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open-variant"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        捲軸模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        橫豎屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="desktop-mac"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        橫屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="desktop-classic"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        豎屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        顯示單章節評論
      </Text>
    </View>
    <ComicBookSwitch />
  </View>
</View>
`;

exports[`Testing ConfigBar component renders correctly with no props 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "height": 667,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": undefined,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        亮度調節
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 2,
        }
      }
    >
      <Icon
        color="white"
        name="decagram"
        type="material-community"
      />
    </View>
    <View
      style={
        Object {
          "flex": 7,
          "paddingTop": 0,
        }
      }
    >
      <ComicBookSlider
        onValueChange={undefined}
        style={
          Object {
            "width": 428.57142857142856,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 2,
        }
      }
    >
      <Icon
        color="white"
        name="decagram-outline"
        type="material-community"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        閱讀模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        普通模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open-page-variant"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        日漫模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="book-open-variant"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        捲軸模式
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        橫豎屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="desktop-mac"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        橫屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    >
      <Icon
        color="white"
        name="desktop-classic"
        type="material-community"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        豎屏
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 0,
          "flex": 1,
          "height": 133.4,
          "justifyContent": "center",
          "marginRight": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        顯示單章節評論
      </Text>
    </View>
    <ComicBookSwitch />
  </View>
</View>
`;
